src/ServerExtension.php

```diff
@@ -83,17 +84,27 @@ public static function takeScreenshots(string $type, string $test): void
         }
 
         foreach (self::$registeredClients as $i => $client) {
-            $screenshotPath = sprintf('%s/%s_%s_%s-%d.png',
-                $_SERVER['PANTHER_ERROR_SCREENSHOT_DIR'],
-                date('Y-m-d_H-i-s'),
-                $type,
-                strtr($test, ['\\' => '-', ':' => '_']),
-                $i
-            );
-            $client->takeScreenshot($screenshotPath);
+            $screenshotPath = self::takeClientScreenshot($_SERVER['PANTHER_ERROR_SCREENSHOT_DIR'], $type, $test, $client, $i);
             if ($_SERVER['PANTHER_ERROR_SCREENSHOT_ATTACH'] ?? false) {
                 printf('[[ATTACHMENT|%s]]', $screenshotPath);
             }
         }
     }
+
+    public static function takeClientScreenshot(string $screenshotDirectory, string $type, string $test, WebDriver $client, int $clientNumber): string
+    {
+        $testSlug = strtr($test, ['\\' => '-', ':' => '_']);
+        $testSlug = substr(preg_replace("/[^-_A-Za-z0-9 ]/", '', $testSlug), 0, 150);
+        $screenshotPath = sprintf('%s/%s_%s_%s_%s-%d.png',
+                                  $screenshotDirectory,
+                                  date('Y-m-d_H-i-s'),
+                                  $type,
+                                  $testSlug,
+                                  md5($test),
```

Comment:
With my new slugification, some tests that are run in the same second could end up with the same filename. For example consider:

- `AcmeTest__testFile with data set ą`
- `AcmeTest__testFile with data set ę`
- `AcmeTest__testFile with data set !`
- `AcmeTest__testFile with data set @`

That last special character would be omitted, so all those tests would have the same name, as long as they run within the same second. 

That's why I'm adding md5 of the `$test` before the slugification to the filename.